### ISSUE TITLE
Option to disable missing_docs for trivial public init's

### ISSUE THREAD

**Issue by bobbradley** [realm/SwiftLint#4107]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

We often need trivial public init functions to allow an object to created. But this causes a missing_docs warning from SwiftLint unless we add doc comments with something generic like "Initializes xyz".

1. Why should this rule be added? Share links to existing discussion about what
It would be nice if there was a way to disable missing_docs for public init functions with no parameters, such as excludes_trivial_init. This would let us keep missing_docs for the other situation.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.
Example where it should not trigger if excludes_trivial_init is set.
```
public init() {}
public init() {}
public init() {
    self.property1 = 123
}

/// Initializes with a name.
public init(name: String) {
    self.name = name
}
```

Example where it should still trigger even if excludes_trivial_init is set.
```
public init(name: String) {
    self.name = name
}
```

3. Should the rule be configurable, if so what parameters should be configurable?
Yes, a configuration option, such as excludes_trivial_init, would be important.

4. Should the rule be opt-in or enabled by default? Why?
Opt-in to avoid causing problems for existing code.
